xbmc/dialogs/GUIDialogContextMenu.cpp

```diff
@@ -493,6 +494,9 @@ bool CGUIDialogContextMenu::OnContextButton(const std::string &type, const CFile
       {
         // don't prompt user for mastercode when reactivating a lock
         g_passwordManager.LockSource(type, share->strName, true);
+
+        // refresh favourites due to possible visibility changes
```

Comment:
The comment should contain why this needs to be called here. Not a general reason but the specific reason.

Reply:
yes, improved all comments and some blemish